backend/fms_core/viewsets/sample_next_step_by_study.py

```diff
@@ -39,9 +40,25 @@ def destroy(self, request, pk=None):
                     sample = Sample.objects.get(id=sample_id)
                     study = Study.objects.get(id=study_id)
                     removed, errors, _ = dequeue_sample_from_specific_step_study_workflow(sample, study, order)
+
+                    if removed:
```

Comment:
I was thinking of doing that but since it's a very small code addition, I assumed it wasn't worth making it it's own function.
I wonder what @UlysseFG thinks of this.

Reply:
It does not go into dequeue... because it is used by the template and the logic is different. 
We weighted the idea of having a wrapper in services but did seem unlikely to see much use. I agree it would be cleaner.
You can wrap this up in a service try to find a name that does not clash with the template ones so it's clear it is for stand-alone dequeue.